### PR TITLE
[Merged by Bors] - chore: adaptations for nightly-2024-08-20

### PR DESCRIPTION
**Pull request by jcommelin**

### CODE REVIEW

File: Mathlib/Data/List/Sort.lean [hunk 1]
```diff
@@ -377,39 +377,39 @@ theorem perm_split : ∀ {l l₁ l₂ : List α}, split l = (l₁, l₂) → l ~
     exact ((perm_split e).trans perm_append_comm).cons a
 
 /-- Implementation of a merge sort algorithm to sort a list. -/
-def mergeSort : List α → List α
+def mergeSort' : List α → List α
```
Comment (fpvandoorn) [inline Mathlib/Data/List/Sort.lean:380]: Is `mergeSort` now upstream? Can you add a comment/todo about the status of this?

**Review by fpvandoorn — APPROVED** [review]
Otherwise LGTM

bors d+

**Comment by semorrison** [comment]
bors merge

File: Mathlib/Data/List/Sort.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Data/List/Sort.lean:398]: I've written a few paragraphs in the module doc.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**